src/prommis/nanofiltration/membrane_cascade_flowsheet/diafiltration_flowsheet_model.py

```diff
@@ -0,0 +1,1120 @@
+                                                    .unfix()
+
+                    propagate_state(
+                        source=getattr(
+                            m.fs.recycle_splitters[i],
+                            f'outlet_{j}'
+                        ),
+                        destination=m.fs.inlet_mixers[i-1, j].recycle
+                    )
+            elif mixing == 'stage':
+                propagate_state(
+                    source=m.fs.split_retentate[i].recycle,
+                    destination=m.fs.inlet_mixers[i-1].recycle
+                )
+
+    def initialize_single_stage(self, m, stage, mixing='tube'):
+        """Initialize a single stage."""
+        # set up initializers
+        stage_initializer = MSContactorInitializer()
+
+        # propagate side stream inlets
+        i = stage
+        for j in pyo.RangeSet(self.nt):
+            m.fs.stage[i].retentate_side_stream_state[0, j].flow_vol.unfix()
+            for sol in self.solutes:
+                m.fs.stage[i].retentate_side_stream_state[0, j]\
+                             .mass_solute[sol]\
+                             .unfix()
+            if mixing == 'tube':
+                propagate_state(
+                    source=m.fs.inlet_mixers[i, j].outlet,
+                    destination=getattr(
+                        m.fs.stage[i],
+                        f'retentate_side_stream_{j}'
+                    )
+                )
+            elif mixing == 'stage':
+                propagate_state(
+                    source=getattr(
+                        m.fs.splitters[i],
+                        f'outlet_{j}'
+                    ),
+                    destination=getattr(
+                        m.fs.stage[i],
+                        f'retentate_side_stream_{j}'
+                    )
+                )
+            m.fs.stage[i].retentate_side_stream_state[0, j].flow_vol.fix()
+            for sol in self.solutes:
+                m.fs.stage[i].retentate_side_stream_state[0, j]\
+                             .mass_solute[sol]\
+                             .fix()
+
+            # the mscontactor initializer sets the initial value of the stream
+            # to the sum of its inlets. Set material transfer to permeate to a
+            # nonzero value to avoid warnings of permeate values being lower
+            # than the LB of 1e-8.
+            m.fs.stage[i].material_transfer_term[0, j, 'permeate', 'retentate',
+                                                 'solvent'].set_value(1e-4)
+            # m.fs.stage[i].permeate[0.0, j].flow_vol.setlb(0)
+            for sol in self.solutes:
+                m.fs.stage[i].material_transfer_term[0, j, 'permeate',
+                                                     'retentate',
+                                                     sol].set_value(1e-4)
+                # m.fs.stage[i].permeate[0.0, j].mass_solute[sol].setlb(0)
```

Comment:
Should these two lines be removed, or uncommented?

Reply:
Yes, deleted.